### PR TITLE
Fix column order in table widget

### LINKED ISSUE #3754: [Bug] Table primary columns incorrect index
## Description

Viewing column properties after specific steps leads to incorrect column properties showing in the property pane panel; in some cases, it causes a crash.

### Steps to reproduce the behaviour:
1. Hide a column in the middle of the list of columns
2. Drag a column over the hidden column and place it before or after the hidden column
3. Try to open a column's properties in the property pane.
4. If the issue doesn't show up yet, try steps 1-3 multiple times
5. See error

### Important Details
Release 1.4.5

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
Fix issue where table column order list gets duplicate values when reordering from the table headers, with a hidden column in between the start and end index of the dragged column

Fixes #3754 #3743 #3552

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manually on JARVIS app

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/propertyControls/PrimaryColumnsControl.tsx [hunk 1]
```diff
@@ -251,6 +251,8 @@ class PrimaryColumnsControl extends BaseControl<ControlProps> {
       this.props.widgetProperties.columnOrder,
     );
 
+    console.log("Table log:", { originalColumn });
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/PrimaryColumnsControl.tsx:254]: Missed the console.log

File: app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx [hunk 2]
```diff
@@ -348,7 +348,7 @@ export const reorderColumns = (
 ) => {
   const newColumnsInOrder: Record<string, ColumnProperties> = {};
 
-  columnOrder.forEach((id: string, index: number) => {
+  uniq(columnOrder).forEach((id: string, index: number) => {
```
Comment (hetunandu) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx:351]: Can we add a jest test for `reorderColumns` function

File: app/client/src/components/designSystems/appsmith/TableComponent/index.tsx [hunk 3]
```diff
@@ -133,9 +133,10 @@ const ReactTableComponent = (props: ReactTableComponentProps) => {
           const columnOrder = props.columnOrder
             ? [...props.columnOrder]
             : props.columns.map((item) => item.accessor);
-          const draggedColumn = props.columns[dragged].accessor;
-          columnOrder.splice(dragged, 1);
-          columnOrder.splice(i, 0, draggedColumn);
+          const movedColumnName = columnOrder.splice(dragged, 1);
+          if (movedColumnName && movedColumnName.length === 1)
+            columnOrder.splice(i, 0, movedColumnName[0]);
+          console.log("Table log:", { columnOrder });
```
Comment (sheelpriy) [inline app/client/src/components/designSystems/appsmith/TableComponent/index.tsx]: console.log ?

Comment (riodeuno) [reply app/client/src/components/designSystems/appsmith/TableComponent/index.tsx]: Hi @sheelpriy Thank's for pointing this out. The console log has been removed.

**Review by mohanarpit — APPROVED** [review]

**Review by kaushik94 — DISMISSED** [review]

**Review by Nikhil-Nandagopal — APPROVED** [review]

**Pull request merged**